rust/xaynet-server/src/services/messages/state_machine.rs

```diff
@@ -12,16 +12,20 @@ use crate::{
 pub use crate::state_machine::{StateMachineError, StateMachineResult as StateMachineResponse};
 
 /// A service that hands the requests to the state machine
-/// ([`crate::state_machine::StateMachine`]) that runs in the
+/// ([`StateMachine`]) that runs in the
 /// background.
+///
+/// [`StateMachine`]: crate::state_machine::StateMachine
 pub struct StateMachineService {
     handle: RequestSender,
 }
 
 impl StateMachineService {
     /// Create a new service with the given handle for forwarding
     /// requests to the state machine. The handle should be obtained
-    /// via [`crate::state_machine::StateMachine::new`]
+    /// via [`StateMachine::new`].
+    ///
+    /// [`StateMachine::new`]: crate::state_machine::StateMachine::new
```

Comment:
Is this actually needed?

Reply:
I thought `StateMachine::new` it looks a bit better than the full path

![Screen Shot 2020-08-31 at 17 12 48](https://user-images.githubusercontent.com/10488408/91735875-376d4900-ebad-11ea-89e9-7891e46e4456.png)

I can change it back if you like